src/utils.jl

```diff
@@ -36,27 +36,27 @@ function buffer_readable(g, limit, ematch_buffer)
   k = length(ematch_buffer)
 
   while k > limit
-    delimiter = ematch_buffer[k]
+    delimiter = ematch_buffer.v[k]
```

Comment:
What's the motivation behind this change? The idea is that operations on the `OptBuffer` ensure correctness as well

Reply:
It seems ```getindex(b::OptBuffer, ...)``` is not implemented. Therefore, we need to access the contained vector directly. This is similar code as in eqsat_apply!() https://github.com/JuliaSymbolics/Metatheory.jl/blob/0b2e6c9f11b9f227f278a058290780f126ee77b6/src/EGraphs/saturation.jl#L220  .

I guess somebody introduced the wrapper type for OptBuffer and forgot to update the ```buffer_readable``` function because it is never called. 

> The idea is that operations on the OptBuffer ensure correctness as well

Not sure what you mean.

Reply:
Ah true I didn't update this function. LGTM then